daisy/instance.go

```diff
@@ -371,13 +373,26 @@ func newInstanceRegistry(w *Workflow) *instanceRegistry {
 	return ir
 }
 
+// SleepFn function is mocked on testing.
+var SleepFn = time.Sleep
+
 func (ir *instanceRegistry) deleteFn(res *Resource) dErr {
 	m := namedSubexp(instanceURLRgx, res.link)
-	err := ir.w.ComputeClient.DeleteInstance(m["project"], m["zone"], m["instance"])
-	if gErr, ok := err.(*googleapi.Error); ok && gErr.Code == http.StatusNotFound {
-		return typedErr(resourceDNEError, err)
+	for i := 1; i < 4; i++ {
+		if _, err := ir.w.ComputeClient.GetInstance(m["project"], m["zone"], m["instance"]); err != nil {
+			// Can't remove an instance that was not even yet created!
+			// However as the command was already submitted, wait.
+			SleepFn((time.Duration(rand.Intn(1000))*time.Millisecond + 1*time.Second) * time.Duration(i))
+			continue
+		}
```

Comment:
Just end the for loop here, always end the function with a DeleteInstance call and whatever error may result